### ISSUE TITLE
Missing text Loco Roco 2 MuiMui House

### ISSUE THREAD

**Issue by Ralayax** [hrydgard/ppsspp#10755]
What happens?

No text appears in the menu when entering the muimui house.

![image](https://user-images.githubusercontent.com/3892807/37661041-20d07796-2c54-11e8-9507-d8406333520c.png)

What *should* happen?
This :

![image](https://user-images.githubusercontent.com/3892807/37661168-6c23b9d8-2c54-11e8-8399-966bee150836.png)

What hardware and operating system are you running PPSSPP on? GPU might matter if it's a graphical issue.

PPSSPP 1.5.4 Official Win64 Build
Windows 10
Ryzen 1700X
AMD RX 580

**Comment by zminhquanz** [comment]
I've test on my PC using NVIDIA GPU , it's doesn't appear that bug , using latest build 1.5.4-720 , I think it caused by AMD drivers
![image](https://user-images.githubusercontent.com/11285640/37664718-72c5e8a8-2c8e-11e8-880e-3721240f6c05.png)

**Comment by Ralayax** [comment]
After some experimenting : 
- language selection doesn't **change** the bug
- with DirectX11 and Vulkan the bug isn't there, only with OpenGL (DirectX9 no tested)

I'm gonna try with Linux, and i'll mess with various AMD Drivers.

I unfortunately don't have any Nvidia GPU.

**Comment by hrydgard** [comment]
It's one of those depth precision things where we are somehow very slightly off and it breaks...

**Comment by hrydgard** [comment]
@Ralayax , an idea: Open assets/compat.ini, and add LocoRoco's game id under [VertexDepthRounding] and/or [PixelDepthRounding] like the other games there. Does that help?

**Comment by zminhquanz** [comment]
@hrydgard Oh , yes
![image](https://user-images.githubusercontent.com/11285640/37891371-5085f0d4-30fe-11e8-92eb-dfdb6f67634d.png)
On OpenGL it's show that bug , and except about lack of driver

**Comment by Ralayax** [comment]
> @Ralayax , an idea: Open assets/compat.ini, and add LocoRoco's game id under [VertexDepthRounding] and/or [PixelDepthRounding] like the other games there. Does that help?

Neither of them fixes the bug for me

**Comment by unknownbrackets** [comment]
Could you try exporting a GE debugger dump on PC?

To do this, open the game and select Debug -> GE debugger..., then when it's displaying the scene, press Record in the top right.  After a second or so, it'll finish and save a trace of the drawing activity.

After that, check the memstick/PSP/SYSTEM/DUMP folder and it'll have created a file named something like "ULES12345_0000.ppdmp".  You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by MasterOfBlack** [comment]
The textures also seem to be broken in Tropuca levels when you have that moving "shrimp" if you are using Vulkan but muimui house works fine. Tropuca level works normally with OpenGL but then the muimui house appears broken (no text).

**Comment by Panderner** [comment]
Even I used real PSP fonts on OpenGL on Android Still No text.
Here's a GE Dump:
[UCUS98731_0001.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/4288277/UCUS98731_0001.ppdmp.zip)

**Comment by hrydgard** [comment]
Yeah I think this is about some depth buffer / viewport issue, not fonts.

**Comment by unknownbrackets** [comment]
For me, that dump shows this in OpenGL:

![image](https://user-images.githubusercontent.com/191233/75953127-88ea8480-5e65-11ea-884f-23bd6d4d63e3.png)

Is that the correct place?  Are you not seeing the same text I see in this screen?

-[Unknown]

**Comment by Panderner** [comment]
> For me, that dump shows this in OpenGL:
> 
> ![image](https://user-images.githubusercontent.com/191233/75953127-88ea8480-5e65-11ea-884f-23bd6d4d63e3.png)
> 
> Is that the correct place? Are you not seeing the same text I see in this screen?
> 
> -[Unknown]

Oops I meant this one:
![Screenshot_2020-03-09-07-38-57-74_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/76173280-1963f480-61d9-11ea-8ddb-28ef3ae20675.png)
[recording.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/4304046/recording.ppdmp.zip)

**Comment by unknownbrackets** [comment]
Text is drawn at 218/223, from a valid looking texture.  It uses depth clamp and a negative Z value (I see -1.8 as the value.)  Z is flat, at least.

Interesting that this displays on AMD without the rounding.

-[Unknown]

**Comment by hrydgard** [comment]
Probably through-mode vs non-throughmode, or two different viewports or something, just on the very edge of failing the depth test, exposing tiny implementation differences...

**Comment by unknownbrackets** [comment]
Well, the Z this sends for that draw is: -1.000025.  The projection matrix flips the sign.

So I think this is only an issue with non-accurate depth.  I think it's getting clipped.

-[Unknown]

**Comment by unknownbrackets** [comment]
We now use accurate depth / depth clamp on OpenGL where supported, and this appears to be working.

It might still not be working on older GL devices, though.  As a note, depth testing is not enabled for this draw, it's just whether it gets clipped or not.

-[Unknown]

**Comment by hrydgard** [comment]
I guess let's leave it open as documentation? Or might as well close, not sure heh.

**Comment by Panderner** [comment]
I tested this game on my PC with NVIDIA GPU it's working fine on all backends in latest builds.
![Screenshot (15)](https://user-images.githubusercontent.com/37503397/149127857-e0379f05-aecc-4bf7-9548-78eeda98c4e7.png)

**Comment by Panderner** [comment]
But the Text still not displaying on my POCO M3 using OpenGL. (Although Vulkan and Desktop OpenGL displays properly)
![Screenshot_2022-01-12-19-00-34-021_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/149128305-72d1265f-89ab-465d-8af9-58e3aeab6a52.jpg)
